source/simulator/stokes_matrix_free.cc

```diff
@@ -2421,6 +2630,14 @@ namespace aspect
       additional_data.mapping_update_flags = (update_values | update_gradients |
                                               update_JxW_values | update_quadrature_points);
 
+      if (!sim.mesh_deformation->get_free_surface_boundary_indicators().empty())
+        additional_data.mapping_update_flags_boundary_faces =
+          (update_values  |
+           update_gradients |
```

Comment:
do we need the gradients?

Reply:
it's copied from matrix-based version, not sure if it's needed in the material model. 
https://github.com/tjhei/aspect/blob/30a5e9658c76e4132e114b5d467f7fc25930455f/source/mesh_deformation/free_surface.cc#L202

Reply:
I don't see where you use derivatives, so try it out without.

Reply:
you are right!